applications/nrf_desktop/src/util/CMakeLists.txt

```diff
@@ -6,6 +6,23 @@
 target_sources_ifdef(CONFIG_DESKTOP_HWID
 		     app PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/hwid.c)
 
+if(CONFIG_HWINFO_NRF AND CONFIG_BOARD_NRF54H20DK_NRF54H20_CPUAPP)
```

Comment:
Could you migrate the HWinfo part as a dedicated commit (as it was originally contributed)? We may want to revert it at some point.

BTW, is this change still needed? If the FICR registers are populated we can remove this workaround.